src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -58,6 +58,8 @@ public virtual FormulaValue GetResult(IRContext irContext)
         private class MinAgg : IAggregator
         {
             protected double _minValue = double.MaxValue;
+            protected DateTime _minValueDT = DateTime.MaxValue;
+            protected TimeSpan _minValueT = TimeSpan.MaxValue;
             protected int _count;
 
             public void Apply(FormulaValue value)
```

Comment:
>FormulaValue [](http://example.com/codeflow?start=30&length=12)

Are we guaranteed that all values are of the same type?

Ie, if somebody does: Min(number, datetime, timespan), it would either be a binding error or the IR would have insert coercions?

Ie, is it possible to call this same aggregator instance with a different types?

Reply:
The only mixed type that is allowed is Date and DateTime, and in that case dates are coerced to DateTime.

Whenever a number is mixed with another type, the other type is coerced to numeric (where possible).

Reply:
In that case, I think we should consider having multiple derived classes per type,  MinDoubleAgg, MinDateTimeAgg, MinTimeSpanAgg.

It's more classes, but each class is trivial and only has a single field and no switching in Apply/GetResult.  Instead, the switching is done once when we create the class instance, and the rest is polymorphism.

Reply:
I created the new Agg classes